### PR TITLE
[timeseries] Fix platform tests failing because of inplace dataframe modifications

### PR DESCRIPTION
**Pull request by shchur**
*Description of changes:*
- Avoid in-place modifying the dataframe, which causes platform tests to fail on certain platforms

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by yinweisu** [comment]
| Previous CI Run | Current CI Run |
| :---: | :---: |

File: .github/workflows/platform_tests-command.yml [hunk 1]
```diff
@@ -243,7 +243,7 @@ jobs:
         if: matrix.os == 'macos-latest'
         shell: bash -l {0}
         run: |
-          wget https://raw.githubusercontent.com/Homebrew/homebrew-core/fb8323f2b170bd4ae97e1bac9bf3e2983af3fdb0/Formula/libomp.rb
+          wget https://raw.githubusercontent.com/Homebrew/homebrew-core/9fab537098b1ee8dc7f4fe10bcbad3305d9e029a/Formula/lib/libomp.rb  # libomp v18.1
```
Comment (canerturkmen) [inline .github/workflows/platform_tests-command.yml:246]: needs to be reverted

**Review by canerturkmen — APPROVED** [review]

**Comment by yinweisu** [comment]
| Previous CI Run | Current CI Run |
| :---: | :---: |
| botocore==1.34.81 | botocore==1.34.82 |
| boto3==1.34.81 | boto3==1.34.82 |
| botocore==1.34.81 | botocore==1.34.82 |
| boto3==1.34.81 | boto3==1.34.82 |

**Pull request merged**